cloud/ec2_client.go

```diff
@@ -868,9 +868,14 @@ func (c *awsClientImpl) CreateFleet(ctx context.Context, input *ec2.CreateFleetI
 			// item describing the error, using an error type that does _not_
 			// implement awserr.Error. We therefore have to check this case in addition
 			// to the standard `err != nil` case above.
-			if len(output.Errors) > 0 {
-				grip.Error(message.WrapError(errors.New(output.Errors[0].String()), msg))
-				return true, err
+			if !ec2CreateFleetResponseContainsInstance(output) {
```

Comment:
I disagree, because for many instance types there is always an error, that they are not supported in a single AZ. Despite this error there is an instance ID, and the host spawns.

It doesn't seem to be the case that when `output.Errors` has a non-zero length that there is an actual error condition. Instead it seems like a nil `err` from `c.EC2.CreateFleetWithContext` combined with a instance ID in `output` means there is not an error. Otherwise there is.

Reply:
I think the reason for the error is that we ask Fleet for a particular instance type and (in the case of a VPC) specify the subnets (and hence the AZs) we want it to choose from. 
https://github.com/evergreen-ci/evergreen/blob/a9698a2138161bb9a93fc606439c9a7292fbefdc/cloud/ec2_fleet.go#L419
Since one of those combinations is invalid AWS returns an instance, but lets us know there was an issue with the way we asked for it. 

If this explanation is correct, we should be able to address the error.

Reply:
alternatively, if it doesn't bother us given that we got an instance one way or another, ignoring the error works also.

Reply:
> If this explanation is correct, we should be able to address the error.

Since it's an undocumented error, I think we should ignore it.